From 07d4f6955dd5dd5535b3e2f8e4722eb20f95e613 Mon Sep 17 00:00:00 2001
From: Renata Ravanelli <rravanel@redhat.com>
Date: Tue, 12 Nov 2024 15:12:52 -0300
Subject: [PATCH] util/osrelease.py:  Replaced string stripping with
 `shlex.split()`

- Replaced string stripping with `shlex.split()` to properly
handle values in the os-release file;
- This ensures cleaner and more accurate key-value assignments,
follwing a broader set of shell-like parsing rules;
- Add os-release file for Fedora CoreOS 40 for testing.

Signed-off-by: Renata Ravanelli <rravanel@redhat.com>
---
 osbuild/util/osrelease.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/osbuild/util/osrelease.py b/osbuild/util/osrelease.py
index b8d56e73..cc97faf5 100644
--- a/osbuild/util/osrelease.py
+++ b/osbuild/util/osrelease.py
@@ -5,6 +5,7 @@ related documentation can be found in `os-release(5)`.
 """
 
 import os
+import shlex
 
 # The default paths where os-release is located, as per os-release(5)
 DEFAULT_PATHS = [
@@ -33,7 +34,10 @@ def parse_files(*paths):
                 if line[0] == "#":
                     continue
                 key, value = line.split("=", 1)
-                osrelease[key] = value.strip('"')
+                split_value = shlex.split(value)
+                if not split_value or len(split_value) > 1:
+                    raise ValueError(f"Key '{key}' has an empty value or more than one token: {value}")
+                osrelease[key] = split_value[0]
 
     return osrelease
 
-- 
2.47.0

